spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/SpatialConvolution.scala

```diff
@@ -173,16 +173,40 @@ class SpatialConvolution[T: ClassTag](
     zeroGradParameters()
   }
 
-  private def getOutputShape(oh: Int, ow: Int, batchSize: Int = -1): Array[Int] = {
+  override def computeOutputShape(inputShape: Shape): Shape = {
+    val input = inputShape.toSingle().toArray
+    require(input.length == 4,
+      "SpatialConvolution: " + ErrorInfo.constrainInputAs3DOrBatch)
+    val (dimHeight, dimWidth, channelDim) = format.getHWCDims(input.length)
+    require(input(channelDim -1) == nInputPlane, s"input channel size " +
+      s"${input(channelDim -1)} is not the same as nInputPlane $nInputPlane")
+    val inputWidth = input(dimWidth -1)
+    val inputHeight = input(dimHeight -1)
+    val sizes =
+      if (padW == -1 && padH == -1) {
+        Utils.getSAMEOutSizeAndPadding(inputHeight, inputWidth, strideH, strideW, kernelH, kernelW)
+      } else {
+        Utils.getOutSizeAndPadding(inputHeight, inputWidth, strideH, strideW,
+          kernelH, kernelW, padH, padW, ceilMode = false)
+      }
+    val outputHeight = sizes(4)
+    val outputWidth = sizes(5)
+    require(outputWidth >= 1 && outputHeight >= 1,
+      s"output size is too small. outputWidth: $outputWidth, outputHeight: $outputHeight")
+    val outputShape = getOutputShape(outputHeight, outputWidth, input(0))
+    Shape(outputShape)
+  }
+
+  private def getOutputShape(oh: Int, ow: Int, batchSize: Int = -2): Array[Int] = {
     format match {
       case DataFormat.NCHW =>
-        if (batchSize == -1) {
+        if (batchSize == -2) {
```

Comment:
add doc here why changing this.

Reply:
ok =  =
-1 means no batch in this method, which conflicts with our inputshape

Reply:
Updated.